lasagne/layers/normalization.py

```diff
@@ -101,3 +115,233 @@ def get_output_for(self, input, **kwargs):
             scale += self.alpha * input_sqr[:, i:i+ch, :, :]
         scale = scale ** self.beta
         return input / scale
+
+
+class BatchNormLayer(Layer):
+    """
+    lasagne.layers.BatchNormLayer(incoming, axes='auto', epsilon=1e-4,
+    alpha=0.1, nonlinearity=None, mode='low_mem',
+    beta=lasagne.init.Constant(0), gamma=lasagne.init.Constant(1),
+    mean=lasagne.init.Constant(0), var=lasagne.init.Constant(1), **kwargs)
+
+    Batch Normalization
+
+    This layer implements batch normalization of its inputs, following [1]_:
+
+    .. math::
+        y = \\frac{x - \\mu}{\\sqrt{\\sigma^2 + \\epsilon}} \\gamma + \\beta
+
+    That is, the input is normalized to zero mean and unit variance, and then
```

Comment:
Ah, right. I'll add a sentence or two explaining what it's normalizing over.

> It is also possible to do ZMUV normalization spatially only, this is sometimes referred to as global contrast normalization.

Hmm, this reminds me that when you're not normalizing over the batch dimension, you don't need running averages. We might want to explain this in the `Notes` section.